neuro_data/static_images/configs.py

```diff
@@ -46,7 +46,22 @@ def get_constraint(self, dataset, stimulus_type, tier=None):
             constraint = constraint & (dataset.tiers == tier)
         return constraint
 
-    def get_loaders(self, datasets, tier, batch_size, stimulus_types):
+    def get_sampler(self, tier, indices):
+        assert tier in ['train', 'validation', 'test', None]
+        if tier == 'train':
+            sampler = SubsetRandomSampler(indices)
+        else:
+            sampler = SubsetSequentialSampler(indices)
+        return sampler
+
+    def log_loader(self, loader):
+        log.info('Loader sampler is {}'.format(loader.sampler.__class__.__name__))
+        log.info('Number of samples in the loader will be {}'.format(len(loader.sampler)))
+        log.info('Number of batches in the loader will be {}'.format(int(np.ceil(len(loader.sampler) / loader.batch_size))))
+
+    def get_loaders(self, datasets, tier, batch_size, stimulus_types, loader_tier):
```

Comment:
I propose to not name that `loader_tier`. Why couldn't you actually just pass in the sampler class? You can override that in you own `load_data` function. Only if `sampler=None` the `get_sampler` function is used. That seems cleaner to me.

Reply:
In the `load_data` you could do `super().load_data(..., sampler=RandomSubsetSampler if tier="train" else SequentialSubsetSampler)`